prime-router/docs/authz/authz-transition-plan.md

```diff
@@ -1,41 +1,136 @@
+For example, an existing user with organization scope `DHmd-phdAdmins` would be added to the `md-phd` group and assigned
+the `admin` scope. New users created after beginning this effort will need to be onboarded with both the new and 
+deprecated scopes.
+
+### Update frontend claims authorization handling and API connections ###
+The frontend is changed to process user authorization based on the claim structure outlined in the design. After this is
+accomplished, the deprecated organization scopes can be retired. All references to the RS API are changed to be directed
+to the auth microservice. 
+
+### Retire API access to the functionapp ###
+Once all senders and users have been provisioned in Okta using the updated structures we can remove network access to
+the functionapp. All communication from senders or the frontend should occur through the auth microservice.
+
+### Implement sender setup API and frontend (tentative) ###
+A proposed API for creating senders entirely within RS is included in the design. All configurations needed for senders
+can be performed from the Okta admin console, so this is not strictly necessary for migration. If desired, the 
+implementation of this API and the related frontend can occur in parallel to the other migration tasks.
+
+# Dev notes #
+These are miscellaneous dev notes that should be considered during the implementation process.
+
 * Migrate all senders in staging
-  * Can existing public keys be transferred to Okta, or will all senders have to be fully reonboarded?
-  * Do we intend to run both auth systems simultaneously?
+    * Can existing public/private keys be transferred to Okta, or will all senders have to be fully reonboarded?
+* Implement sending profile attributes in claims (API > ReportStream API > Claims > create `appSubmit` claim)
+* API changes for new auth flow
+  * Which Azure functions require authorization processes?
+  * Sender onboarded to use auth microservice will have significantly different authorization flow from original API -
+    need strategy for how authorization decisions are made when passing API calls from microservices to functionapp
+    * Could use some sort of translation function to bridge new authorization to old in order to support legacy
+      endpoint before all senders are reonboarded
+    * Alternative would be to handle all authorization on auth microservice and bypass authorization on functionapp
+      end when auth microservice is the caller 
+* Should sender groups be stored as Okta groups or solely as profile attributes?
+    * Utilizing groups permits onboarding via Okta admin console, but the group memberships are redundant as it is only
+      the profile attributes that are actively used during the authz process
+      * Event hook strategy should keep group memberships and profile attributes in sync without much upkeep 
+    * Storing solely as profile attributes would require the sender management APIs to be built before transition to the
```

Comment:
But if we do both methods then we still need the event hook based updater, right?